scipy/special/xsf/legendre.h

```diff
@@ -21,8 +21,9 @@ struct legendre_p_recurrence_n {
     T z;
 
     void operator()(int n, T (&res)[2]) const {
-        T fac0 = -T(n - 1) / T(n);
-        T fac1 = T(2 * n - 1) / T(n);
+        using value_type = remove_dual_t<T>;
+        value_type fac0 = -value_type(n - 1) / value_type(n);
+        value_type fac1 = value_type(2 * n - 1) / value_type(n);
```

Comment:
I confirmed that all these formulas are the same; we've just made the arguments regular numbers instead of duals numbers.